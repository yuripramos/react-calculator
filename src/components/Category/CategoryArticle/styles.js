import styled from "styled-components";
import { rem } from "../../../styles/tools";
import { grey10, red10, black, darkestBlue } from "../../../styles/settings";
import { Link } from "react-router-dom";

export const ContentWrapper = styled.div`
  text-align: center;
  padding: ${rem(15)} ${rem(10)};
  min-height: ${rem(250)};
  width: 100%;
  display: inline-block;
`;

export const CategoryWrapper = styled.div`
  text-align: left;
  padding: ${rem(15)} ${rem(5)};
  min-height: ${rem(100)};
  display: inline-flex;
  flex-wrap: wrap;
  border-bottom: solid 1px ${grey10};
  width: 100%;
  :nth-child(2n) {
    background: rgba(229, 229, 232, 0.1);
  }
`;
export const Clicker = styled(Link)`
  text-decoration: none;

  &:focus,
  &:hover,
  &:visited,
  &:link,
  &:active {
    text-decoration: none;
  }
  color: ${black};
  font-size: ${rem(30)};
  font-weight: 200;
  line-height: 0.75;
  font-family: "CrimsomText SemiBold";
  margin-top: ${rem(20)};
  width: 100%;
`;

export const Media = styled.img`
  width: ${rem(400)};
  height: ${rem(300)};

`;

export const Notice = styled.div`
  font-family: "OpenSans Regular";
  font-size: ${rem(20)};
  color: ${red10};
  text-align:center;
  height: ${rem(50)};
  display: inline-flex;
  line-height: 45px;
  margin: ${rem(5)};
`;

export const Title = styled.span`
  font-size: ${rem(30)};
  font-weight: 200;
  line-height: 0.75;
  font-family: "CrimsomText SemiBold";
  margin-top: ${rem(20)};
  width: 100%;
`;

export const SubTitle = styled.span`
  margin-top: ${rem(30)};
  font-family: "OpenSans Regular";
  font-weight: normal;
  font-size: ${rem(16)};
  color: ${darkestBlue};
`;


export const PublishedDate = styled.span`
  font-family: "OpenSans Regular";
  font-size: .35em;
  margin: 1.625rem 0;
  letter-spacing: 0.05em;
  color: ${red10};
  width: 100%;
  font-size: .3em;
  margin: ${rem(20)} 0 ${rem(15)} 0;
`;

export const Author = styled.span`
  font-size: .4em;
  line-height: 0.75;
  font-family: "OpenSans Thin";
  margin: ${rem(20)} 0 ${rem(10)} 0;
  width: 100%;
`;
