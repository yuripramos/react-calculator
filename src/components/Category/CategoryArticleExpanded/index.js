import CategoryArticleExpanded from "./CategoryArticleExpanded";
import { withRouter } from "react-router-dom";
import { connect } from "redux-zero/react";
import { combineActions } from "redux-zero/utils";

import modalActions from "../../../actions/modal";
import newsActions from "../../../actions/News";

export default connect(
  ({ topUserStories, isFilled, FeedArray, current }) => ({
    topUserStories,
    isFilled,
    FeedArray,
    current
  }),
  combineActions(newsActions, modalActions)
)(withRouter(CategoryArticleExpanded));
