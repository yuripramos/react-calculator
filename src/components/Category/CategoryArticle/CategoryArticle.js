import React, { Component  } from "react";
import moment from "moment";
import Icon from "../../common/Icon";
import { Container, Row, Column } from "../../../styles/grid";
import { string, arrayOf, shape, bool } from "prop-types";

import {
  ContentWrapper,
  Title,
  CategoryWrapper,
  PublishedDate,
  Author,
  Media,
  Clicker,
  Notice,
  SubTitle
} from "./styles";

class CategoryContent extends Component {
  constructor(props) {
    super(props);
    this.getKeyFromShortenUrl = this.getKeyFromShortenUrl.bind(this);
  }
  getKeyFromShortenUrl(hash) {
    const hashIdFull = hash.split("/");
    const shortHash = hashIdFull[hashIdFull.length - 1];
    if(shortHash.includes(".html")) {
      return shortHash.replace(".html", "");
    }
    return shortHash;
  }

  render() {
    const { content, match, isMainPage } = this.props;
    return (
      <ContentWrapper>
        <Container>
          <Row>
            <Column>
              {isMainPage && (
                <Notice>
                  <Icon name="Attention" width="50px" height="50px" />
                  This page only list the most recent news from all
                  sections, to navigate choose the sections above
                </Notice>
              )}
              {content.map((e, i) => (
                <CategoryWrapper key={`article-${e.section}-${i}`}>
                  {isMainPage && <SubTitle>in {e.section} </SubTitle>}
                  {isMainPage ? <Title>{e.title} </Title> : (
                    <Clicker
                      to={`${match.path}/${this.getKeyFromShortenUrl(
                        e.short_url || e.url
                      )}`}
                    >
                      <Title>{e.title} </Title>
                    </Clicker>
                  )}
                  <PublishedDate>
                    updated in: {moment(e.published_date).format("LLL")}
                  </PublishedDate>
                  {e.multimedia.length > 0 && (
                    <Media
                      src={
                        e.multimedia.length > 4
                          ? e.multimedia[4].url
                          : e.multimedia[2].url
                      }
                      alt="thumbnail"
                    />
                  )}
                  <Author>{e.byline} </Author>
                </CategoryWrapper>
              ))}
            </Column>
          </Row>
        </Container>
      </ContentWrapper>
    );
  }
}
CategoryContent.defaultProps = {};

CategoryContent.propTypes = {
  content: arrayOf(
    shape({
      section: string,
      abstract: string,
      url: string,
      published_date: string,
      byline: string,
      title: string
    })
  ),
  match:
    shape({
      path: string,
    }),
  isMainPage: bool,
};

export default CategoryContent;
