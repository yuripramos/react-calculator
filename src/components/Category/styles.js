import styled, { css } from "styled-components";
import { rem, media } from "../../styles/tools";
import { trueBlack } from "../../styles/settings";

export const ContentWrapper = styled.div`
  text-align: center;
  padding: ${rem(15)} ${rem(10)};
  min-height: ${rem(250)};

  ${media.md(css`
    min-height: ${rem(300)};
    padding: ${rem(20)} ${rem(10)};
  `)};
`;
export const Title = styled.h1`
  font-family: "RobotoSlab Regular";
  font-size: ${rem(30)};
  color: ${trueBlack};
  margin-top: 0;
`;
