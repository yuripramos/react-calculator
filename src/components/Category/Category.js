import React, { Component } from "react";

import CategoryArticle from "./CategoryArticle";
import { Container, Row, Column } from "../../styles/grid";
import { string, arrayOf, shape, func, bool, oneOfType } from "prop-types";
import LocalLoading from "../common/LocalLoading";

import { ContentWrapper, Title } from "./styles";
class Category extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: true
    };
    this._isMounted = false;
  }
  componentWillReceiveProps(){
    const { isFilled } = this.props;
    const { isLoading } = this.state;
    if (isFilled && isLoading) {
      this.setState({
        isLoading: false
      });
    }
  }

  componentDidMount() {
    this._isMounted = true;
    window.onpopstate = () => {
      if (this._isMounted) {
        this.setState({
          isLoading: false
        });
      }
    }
  }

  render() {
    const { isFilled, contextTitle, FeedArray, current, match } = this.props;
    const { isLoading } = this.state;
    isFilled && this.props.filterByType(FeedArray, contextTitle);
    const isHotNewsActive = match.path.includes("hot");
    return (
      <ContentWrapper>
        {isFilled && (
          <Container>
            <Row>
              <Column>
                <Title>
                  {isLoading ? (
                    <LocalLoading />
                  ) : (
                    <CategoryArticle
                      content={
                        isHotNewsActive
                          ? current.results.slice(0, 6)
                          : current.results
                      }
                    />
                  )}
                </Title>
              </Column>
            </Row>
          </Container>
        )}
      </ContentWrapper>
    );
  }
}

  Category.defaultProps = {
    current: false,
  };

Category.propTypes = {
  filterByType: func,
  contextTitle: string,
  isFilled: bool,
  FeedArray: arrayOf(
    shape({
      results: arrayOf(
        shape({
          section: string,
          abstract: string,
          url: string,
          published_date: string,
          byline: string,
          title: string
        })
      )
    })
  ),
  current: oneOfType([
    bool,
    shape({
      results: arrayOf(
        shape({
          section: string,
          abstract: string,
          url: string,
          published_date: string,
          byline: string,
          title: string
        })
      )
    })
  ]),
  match: shape({
    path: string
  })
};

export default Category;
