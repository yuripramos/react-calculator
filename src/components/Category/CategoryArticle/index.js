import CategoryArticle from "./CategoryArticle";
import { withRouter } from "react-router-dom";

export default withRouter(CategoryArticle);
