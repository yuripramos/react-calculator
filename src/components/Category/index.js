import { connect } from "redux-zero/react";

import Category from "./Category";
import newsActions from "../../actions/News";

export default connect(
  ({ isFilled, FeedArray, filterByType, current }) => ({
    isFilled,
    FeedArray,
    filterByType,
    current
  }),
  newsActions
)(Category);
