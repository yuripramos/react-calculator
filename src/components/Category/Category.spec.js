import React from "react";
import { shallow, mount } from "enzyme";
import { spy } from "sinon";
import Category from "./Category";
import CategoryArticle from "./CategoryArticle";

let props;

beforeEach(() => {
  props = {
    FeedArray: [
      {
        results: [
          {
            section: "science",
            abstract: "while ultraviolete fluor",
            url: "http://www.nytimes.com/2019/02/01/science/pink-glow",
            published_date: "2019-02-01T12:42:18-05:00",
            byline: "By VERONIQUE GREENWOOD",
            title: "Flying Squirrels That Glow Pink in the Dark"
          },
          {
            section: "science",
            abstract: "while usadasltraviolete fluor",
            url: "http://www.nytimes.com/2019/02/01/science/red-glow",
            published_date: "2019-02-01T12:42:16-05:00",
            byline: "By BARBARIAN",
            title: "Pink is the new white"
          }
        ]
      },
      {
        results: [
          {
            section: "science",
            abstract: "while ultraviolete fluor",
            url: "http://www.nytimes.com/2019/02/01/science/pink-glow",
            published_date: "2019-02-01T12:42:18-05:00",
            byline: "By VERONIQUE GREENWOOD",
            title: "Flying Squirrels That Glow Pink in the Dark"
          },
          {
            section: "science",
            abstract: "while usadasltraviolete fluor",
            url: "http://www.nytimes.com/2019/02/01/science/red-glow",
            published_date: "2019-02-01T12:42:16-05:00",
            byline: "By BARBARIAN",
            title: "Pink is the new white"
          }
        ]
      }
    ],
    current: {
      results: [
        {
          section: "science",
          abstract: "while ultraviolete fluor",
          url: "http://www.nytimes.com/2019/02/01/science/pink-glow",
          published_date: "2019-02-01T12:42:18-05:00",
          byline: "By VERONIQUE GREENWOOD",
          title: "Flying Squirrels That Glow Pink in the Dark"
        },
        {
          section: "science",
          abstract: "while usadasltraviolete fluor",
          url: "http://www.nytimes.com/2019/02/01/science/red-glow",
          published_date: "2019-02-01T12:42:16-05:00",
          byline: "By BARBARIAN",
          title: "Pink is the new white"
        }
      ]
    },
    match: {
      path: "/science/hot"
    },
    isFilled: true,
    filterByType: jest.fn()
  };
});

describe("Category Component", () => {
  it("should match snapshot with props", () => {
    expect(shallow(<Category {...props} />)).toMatchSnapshot();
  });
  it("should call filterByType", () => {
    const wrapped = mount(<Category {...props} />);
    expect(wrapped.state()).toEqual({ isLoading: true });
    expect(props.filterByType).toHaveBeenCalled();
  });
  it("should render CategoryArticle Comp", () => {
    spy(Category.prototype, "componentWillReceiveProps");
    spy(Category.prototype, "componentDidMount");
    const wrapper = shallow(<Category {...props} />);
    wrapper.setState({ isLoading: false }, () => {
      wrapper.update();
      expect(Category.prototype.componentDidMount.calledOnce);
      expect(Category.prototype.componentWillReceiveProps.calledOnce);
      expect(wrapper.find(CategoryArticle).length).toBe(1);
    });
  });
});
