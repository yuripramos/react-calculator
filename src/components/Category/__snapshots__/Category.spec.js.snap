// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Component should match snapshot with props 1`] = `
<styled.div>
  <Container>
    <Row>
      <Column>
        <styled.h1>
          <LocalLoading />
        </styled.h1>
      </Column>
    </Row>
  </Container>
</styled.div>
`;
